levelkdev/swapr-dapp
# Feature/socket add polygon support

# Summary

Closes #1027

- added polygon token lists to Socket
- socket token list is now fetched from https://github.com/SwaprDAO/swapr-ecobridge-socket-lists
- refactored token substitution for socket ( Ethereum native => WETH Gnosis etc.)
- ecoBridge `fetchDynamicList()` now waits for `init()` to complete

To learn more how socket token lists are created please visit https://github.com/SwaprDAO/swapr-ecobridge-socket-lists

# To Test
On `ecoBridge` select given chain pair and pick currency on the left of the 'table'. 
After that enter some amount (usually 100 is enough) and wait for quote (on routes Socket should be in loading state until quote returns). There you can find `fromAsset` and `toAsset`.
Native address in socket is `0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE`).

Make sure that following routes are found and fromAsset/toAsset are as follows:

(Arbitrum/Ethereum-Gnosis): native => WETH
(Arbitrum/Ethereum-Gnosis): DAI => native

(Gnosis-Arbitrum/Ethereum): native => DAI
(Gnosis-Arbitrum/Ethereum): WETH => native

(Arbitrum-Polygon): native => WETH

(Polygon-Arbitrum): WETH => native

(Ethereum-Polygon): native => WETH 
(Ethereum-Polygon): MATIC => native

(Polygon-Ethereum): native => MATIC //try bigger values if route not found 
(Polygon-Ethereum): WETH => native

(Polygon-Gnosis): native => MATIC
(Polygon-Gnosis): DAI => native

(Gnosis-Polygon): native => DAI
(Gnosis-Polygon): MATIC => native

Make sure that following routes are NOT found:

(Arbitrum-Polygon): MATIC => native //should be no MATIC token to select

(Polygon-Arbitrum): native => MATIC // should be no route

For the rest of the tokens:
Pick random tokens and see if route is found for tokens, and quote is providing correct fromAsset and toAsset (eg. fromAsset is USDC on chainA and toAsset is USDC on chainB)
